src/mpid/ch4/src/ch4r_init.c

```diff
@@ -17,11 +17,8 @@ int MPIDIG_init_comm(MPIR_Comm * comm)
     if (MPIR_CONTEXT_READ_FIELD(DYNAMIC_PROC, comm->recvcontext_id))
         goto fn_exit;
 
-    subcomm_type = MPIR_CONTEXT_READ_FIELD(SUBCOMM, comm->recvcontext_id);
-    is_localcomm = MPIR_CONTEXT_READ_FIELD(IS_LOCALCOMM, comm->recvcontext_id);
-
-    MPIR_Assert(subcomm_type <= 3);
-    MPIR_Assert(is_localcomm <= 1);
+    MPIR_Assert(MPIR_CONTEXT_READ_FIELD(SUBCOMM, comm->recvcontext_id) <= 3);
+    MPIR_Assert(MPIR_CONTEXT_READ_FIELD(IS_LOCALCOMM, comm->recvcontext_id) <= 1);
```

Comment:
I think these asserts can just be removed. There used to be a context_id->comm lookup in this places that was removed.

Reply:
Gone.